packages/core/src/browser/preferences/preference-configurations.ts

```diff
@@ -35,7 +35,7 @@ export class PreferenceConfigurations {
     protected readonly provider: ContributionProvider<PreferenceConfiguration>;
 
     /* prefer Theia over VS Code by default */
-    getPaths(): string[] {
+    async getPaths(): Promise<string[]> {
```

Comment:
Can we keep it sync and provide paths via DI async bindging to avoid breaking APIs? btw `getPaths` still has hardcoded `.theia`

Reply:
Oops. I reworked it to async to be able to use the service, but forgot to finalize. My bad, thanks for spotting.

Reply:
@akosyakov I looked at [async provider](https://github.com/inversify/InversifyJS/blob/master/wiki/provider_injection.md) as you mentioned [here](https://github.com/eclipse-theia/theia/pull/6650#discussion_r359206948), but this is not gonna work in this case (as we need `getPaths` on start, so `null` value will not work).

Also I've tried to do the following:
```typescript
bind(DATA_FOLDER_NAME).toDynamicValue(async ctx => {
    const envServer = <EnvVariablesServer>ctx.container.get(EnvVariablesServer);
    return envServer.getDataFolderName();
});
```
```typescript
@inject(DATA_FOLDER_NAME)
protected readonly DATA_FOLDER_NAME: string;
```
But that returns `{}` instead of string real value.

Do you have any idea how we can keep `getPaths` synchronous?

Reply:
Let's try to keep it async.